src/include/distributed/metadata_cache.h

```diff
@@ -38,11 +38,12 @@ typedef struct
 
 	/* pg_dist_shard metadata (variable-length ShardInterval array) for this table */
 	int shardIntervalArrayLength;
-	ShardInterval *shardIntervalArray;
+	ShardInterval **sortedShardIntervalArray;
```

Comment:
In the code, whenever we needed shard interval arrays, we use array of shard interval pointers instead of array of shard intervals. So, having that in the cache makes its utilization in the code smoother. I think having slower binary search is an acceptable trade of. Is that OK for you?